src/hooks/useAffiliatesInfo.ts

```diff
@@ -38,11 +50,31 @@ export const useAffiliatesInfo = (dydxAddress?: string) => {
     }
   };
 
-  const query = useQuery({
+  const affiliateMetadataQuery = useQuery({
     queryKey: ['affiliatesMetadata', dydxAddress],
-    queryFn,
+    queryFn: fetchAffiliateMetadata,
     enabled: Boolean(compositeClient && dydxAddress),
   });
 
-  return query;
+  const fetchAffiliateMaxEarning = async () => {
+    const taker3FeeTier = feeTiers?.[2].taker ?? DEFAULT_TAKER_3_FEE;
+    const allAffiliateTiers = await getAllAffiliateTiers();
+    let maxRevshare = DEFAULT_MAX_AFFILIATE_SHARE;
+    if (allAffiliateTiers?.tiers?.tiers?.length) {
```

Comment:
its kind of weird that you have to do `.tiers.tiers` from this code - what if the compositeClient just returns the inner field already?

```
const getAllAffiliateTiers = useCallback(async () => {
    return compositeClient?.validatorClient.get.getAllAffiliateTiers()?.tiers?.tiers;
  }, [compositeClient]);
```

Reply:
it also feels like we may want to update `v4-clients` API to be something more like
`getAllAffiliateTiers().data.tiers` unless there's a specific reason we need 2 nested `tier` keys.

Reply:
This typing is from the protocol, so we can't really update it. but its definitely wierd agree. I can strip out a layer in `getAllAffiliateTiers` function.